### PR TITLE
Add --verbose switch and fix some style issues and typos

### PR DESCRIPTION
**Pull request by Flamefire**
Calling `python -m scorep --verbose ...` will print out the effective Score-P config used and the LD_* variables set so the program can also be started manually for testing purposes.
Most useful for use with --keep-files

Example: 
```
python -m scorep --noinstrumenter --cuda --verbose --keep-files exit-region-same-but-not.py
scorep: Score-P config: ['--cuda', '--thread=pthread', '--compiler', '--user']
scorep: Score-P files are kept at: /tmp/scorep.b0if5w3_
scorep: LD_LIBRARY_PATH="/tmp/scorep.b0if5w3_:$LD_LIBRARY_PATH"
scorep: LD_PRELOAD="/sw/installed/Score-P/6.0-gompic-2019b/lib/libscorep_adapter_compiler_event.so /sw/installed/Score-P/6.0-gompic-2019b/lib/libscorep_adapter_user_event.so /sw/installed/Score-P/6.0-gompic-2019b/lib/libscorep_adapter_opencl_event_linktime.so /sw/installed/Score-P/6.0-gompic-2019b/lib/libscorep_adapter_pthread_event.so /sw/installed/Score-P/6.0-gompic-2019b/lib/libscorep_measurement.so /sw/installed/Score-P/6.0-gompic-2019b/lib/libscorep_constructor.so libscorep_init_subsystem-3.7.so"
```

### CODE REVIEW

File: scorep/__main__.py [hunk 1]
```diff
@@ -19,17 +20,22 @@ def scorep_main(argv=None):
...
     no_default_threads = False
     no_default_compiler = False
     no_instrumenter = False
     instrumenter_type = "profile"
 
     for elem in argv[1:]:
         if parse_scorep_commands:
-            if elem == "--mpi":
+            if elem == "--":
+                parse_scorep_commands = False
```
Comment (AndreasGocht) [inline scorep/__main__.py:32]: Hmmm ... why did you do this explicit?

File: scorep/helper.py [hunk 2]
```diff
@@ -74,15 +78,15 @@ def add_to_ld_library_path(path):
         os.environ["LD_LIBRARY_PATH"] = ':'.join([path] + library_paths)
 
 
-def generate_compile_deps(config=[]):
+def generate_compile_deps(config):
     """
     Generates the data needed for compilation.
     """
 
-    scorep_config = ["scorep-config"] + config + ["--user"]
```
Comment (AndreasGocht) [inline scorep/helper.py:82]: Why did you remove the `--user`?

**Review by AndreasGocht — COMMENTED** [review]
I am unsure about the changes to `--user`. Do you have a short rationale here?
Same for adding `--`?

File: scorep/__main__.py (hunk 1 above)
Comment (Flamefire) [reply scorep/__main__.py:32]: This is to allow the usual `--` syntax to stop argument parsing for commands running commands. E.g.: `srun --ntasks=1 -- foobar --help`
There might be users used to that and it is a 2-line addition, so why not?

File: scorep/helper.py (hunk 2 above)
Comment (Flamefire) [reply scorep/helper.py:82]: I moved that up to reduce the amount of magic scorep-python is doing. This makes this helper function more flexible as the caller is responsible for adding the `--user`, see https://github.com/score-p/scorep_binding_python/pull/122/files#diff-33e0c1512b38043a56ab9c6c83e25c7c4c7c9a50283ea20b70f9e645343eafd6R117-R118

This change basically resulted from wanting to print the actual config used at https://github.com/score-p/scorep_binding_python/pull/122/files#diff-33e0c1512b38043a56ab9c6c83e25c7c4c7c9a50283ea20b70f9e645343eafd6R121

If the sub-functions add another argument, then the print is incomplete. So not touching the config afterwards ensures that the output is correct/complete and avoids the code duplication by having to add `--user` in multiple places

File: scorep/__main__.py (hunk 1 above)
Comment (AndreasGocht) [reply scorep/__main__.py:32]: alright. I thought about it. I do agree.

File: scorep/helper.py (hunk 2 above)
Comment (AndreasGocht) [reply scorep/helper.py:82]: Alright.
